proto/regen/ecocredit/basket/v1alpha1/tx.proto

```diff
@@ -0,0 +1,118 @@
+  // as one unit in user interfaces.
+  uint32 exponent = 4;
+
+  // disable_auto_retire allows auto-retirement to be disabled.
+  // The credits will be auto-retired if disable_auto_retire is
+  // false unless the credits were previously put into the basket by the
+  // address picking them from the basket, in which case they will remain
+  // tradable.
+  bool disable_auto_retire = 5;
+
+  // credit_type_name filters against credits from this credit type name.
+  string credit_type_name = 6;
+
+   // allowed_classes are the credit classes allowed to be put in the basket
+  repeated string allowed_classes = 7;
+
+  // batch_date_window is a duration of time measured into the past which sets
+  // a cutoff for batch start dates when adding new credits to the basket.
+  // Based on the current block timestamp, credits whose start date is before
+  // block timestamp - batch_date_window will not be allowed into the basket.
+  google.protobuf.Duration batch_date_window = 8;
+
+  // multiplier is an integer number which is applied to credit units when
+  // converting to basket units. For example if the multiplier is 2000, then
+  // 1.1 credits will result in 2200 basket tokens. If there are any fractional
+  // amounts left over in this calculation when adding credits to a basket,
+  // those fractional amounts will not get added to the basket.
+  string multiplier = 9;
+}
+
+// MsgCreateBasketResponse is the Msg/CreateBasket response type.
+message MsgCreateBasketResponse {
+
+  // basket_denom is the unique denomination ID of the newly created basket.
+  string basket_denom = 1;
+}
+
+// MsgAddToBasket is the Msg/AddToBasket request type.
+message MsgAddToBasket {
+  // owner is the owner of credits being added to the basket.
+  string owner = 1;
+
+  // basket_denom is the basket denom to add credits to.
+  string basket_denom = 2;
+
+  // credits are credits to add to the basket. If they do not match the basket's
+  // admission criteria the operation will fail.
+  repeated BasketCredit credits = 3;
+}
+
+// MsgAddToBasketResponse is the Msg/AddToBasket response type.
+message MsgAddToBasketResponse {
+
+  // amount_received is the amount of basket tokens received.
+  string amount_received = 1;
+}
+
+// MsgTakeFromBasket is the Msg/TakeFromBasket request type.
+message MsgTakeFromBasket {
+
+  // owner is the owner of the basket tokens.
+  string owner = 1;
+
+  // basket_denom is the basket denom to take credits from.
+  string basket_denom = 2;
+
+  // amount is the number of credits to take from the basket.
+  string amount = 3;
+
+  // retirement_location is the optional retirement location for the credits
+  // which will be used only if retire_on_take is true for this basket.
+  string retirement_location = 4;
+}
+
+// MsgTakeFromBasketResponse is the Msg/TakeFromBasket response type.
+message MsgTakeFromBasketResponse {
+
+  // credits are the credits taken out of the basket.
+  repeated BasketCredit credits = 1;
```

Comment:
`BasketCredit`s only have a field for "tradable_amount", but sometimes these credits may actually be retired (if `retired_on_take` is true).

Seems like the easiest solution is maybe to change the `BasketCredit` message to have the field named `amount`instead of `tradable_amount`.